util/types/compare.go

```diff
@@ -13,6 +13,9 @@
 
 package types
 
+// floatEpsilon is the acceptable error quantity when comparing two float numbers.
+const floatEpsilon float64 = 2.2204460492503131e-16
```

Comment:
IEEE 754 defines comparisons between float point numbers, we'd better to check whether golang has already follow the standard in operator `<`, `>`, `==`